apps/nextjs-example/components/transactionFlow/Sponsor.tsx

```diff
@@ -0,0 +1,134 @@
+import { AccountAuthenticator, AnyRawTransaction } from "@aptos-labs/ts-sdk";
+import { useWallet } from "@aptos-labs/wallet-adapter-react";
+import { useState, useEffect } from "react";
+import { aptosClient } from "../../utils";
+import { useAlert } from "../AlertProvider";
+import Button from "../Button";
+import Col from "../Col";
+import Row from "../Row";
+export const APTOS_COIN = "0x1::aptos_coin::AptosCoin";
+type SponsorTransactionProps = {
+  isSendableNetwork: (connected: boolean, network?: string) => boolean;
+};
+
+export default function SponsorTransaction({
+  isSendableNetwork,
+}: SponsorTransactionProps) {
+  const { connected, account, network, signTransaction, submitTransaction } =
+    useWallet();
+  const [transactionToSubmit, setTransactionToSubmit] =
+    useState<AnyRawTransaction | null>(null);
+  const { setSuccessAlertHash } = useAlert();
+
+  const [senderAuthenticator, setSenderAuthenticator] =
+    useState<AccountAuthenticator>();
+  const [feepayerAuthenticator, setFeepayerAuthenticator] =
+    useState<AccountAuthenticator>();
+
+  let sendable = isSendableNetwork(connected, network?.name);
+
+  useEffect(() => {
+    if (!sendable) return;
+    if (!account) return;
+    if (!network) return;
+
+    // Generate a raw transaction using the SDK
+    const generate = async (): Promise<AnyRawTransaction> => {
+      const transactionToSign = await aptosClient(
+        network?.name.toLowerCase()
+      ).generateTransaction({
+        sender: account.address,
+        hasFeePayer: true,
+        data: {
+          function: "0x1::coin::transfer",
+          typeArguments: [APTOS_COIN],
+          functionArguments: [account.address, 1], // 1 is in Octas
+        },
+      });
+      return transactionToSign;
+    };
+
+    // Fund current connected account to create it on chain
+    const fundCurrentAccount = async () => {
+      await aptosClient(network.name.toLowerCase()).fundAccount({
+        accountAddress: account.address,
+        amount: 100_000_000,
+      });
+    };
+
+    // We fund the current connected account to create it on chain
+    // Then we generate the transaction with the current connected account
+    // as the sender
+    fundCurrentAccount().then(() => {
+      generate().then((transactionToSign) =>
+        setTransactionToSubmit(transactionToSign)
+      );
+    });
+  }, [network, account, sendable]);
+
+  const onSignTransaction = async (asSponsor?: boolean) => {
+    if (!transactionToSubmit) {
+      throw new Error("No Transaction to sign");
+    }
+    try {
+      const authenticator = await signTransaction(transactionToSubmit);
+      if (asSponsor) {
+        setFeepayerAuthenticator(authenticator);
+      } else {
+        setSenderAuthenticator(authenticator);
+      }
+    } catch (error) {
+      console.error(error);
```

Comment:
We may want to approach the fee payer as a different account to show the example, similar to multi-agent.  But, this works for now.